build/shared/examples/04.Communication/SerialPassthrough/SerialPassthrough.ino

```diff
@@ -22,12 +22,24 @@
   by Erik Nyquist
 */
 
+uint32_t currentBaud = 9600;
+
 void setup() {
-  Serial.begin(9600);
-  Serial1.begin(9600);
+  while (!Serial) {
+    ; // wait for serial port to connect. Needed for native USB
+  }
+  currentBaud = Serial.baud();
+  Serial1.begin(currentBaud);    // Set hardware serial baud rate to USB's desired baud rate
 }
 
 void loop() {
+  // Poll for baud changes on the USB CDC side
+  if (currentBaud != Serial.baud()) {
+    currentBaud = Serial.baud();
+    Serial1.flush();             // Finish sending any pending messages
+    Serial1.begin(currentBaud);  // Set hardware serial baud rate to USB's desired baud rate
+  }
```

Comment:
If desired, this part could be removed or moved to a more advanced example. Most built projects that just need USB<>UART won't change baud at runtime, but this is necessary to fully imitate an FTDI-like USB<>UART board.

Reply:
I think this might need to call end and then begin (though maybe just calling begin again might work). I wrote a similar sketch a long time ago for the Leonardo and used end and begin, but had to keep the TX pin HIGH to prevent a glitch.

See: https://www.stderr.nl/static/files/Hardware/Electronics/Arduino/XBee/LeonardoForward.ino